packages/vite/src/node/plugins/css.ts

```diff
@@ -803,7 +806,7 @@ async function compileCSS(
   const needInlineImport = code.includes('@import')
   const hasUrl = cssUrlRE.test(code) || cssImageSetRE.test(code)
   const lang = id.match(CSS_LANGS_RE)?.[1] as CssLang | undefined
-  const postcssConfig = await resolvePostcssConfig(config, getCssDialect(lang))
+  const postcssConfig = await resolvePostcssConfig(config)
```

Comment:
Dialect option can be removed as it was only separated because sugarss mutates the config to add the sugarss parser. Down the code I've refactored to not mutate the config so we can keep a single cache as before.